---
title: "Normalization of the different studies"
author: "Mar Batlle"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = TRUE,
	message = FALSE,
	warning = FALSE
)
library(tidyverse)
library(readr)
library(DESeq2)
library(apeglm)
library(RobustRankAggreg)
library(BiocParallel)
register(MulticoreParam(2))
```


# Import data
Count matrix should not be normalized and should be uniquely using hgnc gene symbols.
```{r}
## Import Raw counts matrix
cts_path <- "~/Documents/TFM/GitHub/HB_PublicData/1-Obtaining-DEGs-for-HB/Matrices_HB/Originals_HB/GSE104766_raw_data.csv"
cts <- read.csv(cts_path, row.names=1)
```

```{r}
## Import Metadata
meta_path <- '~/Documents/TFM/GitHub/HB_PublicData/1-Obtaining-DEGs-for-HB/Metadata_HB/GSE104766_metadata.csv'
coldata <- read.csv(meta_path, row.names=1)
coldata$tissue <- factor(coldata$tissue)
```


## Find DEGs
```{r}
# Run differential expression analysis
dds <- DESeq(dds, parallel=TRUE)
res <- results(dds, lfcThreshold = 1, parallel=TRUE)

# Extract all differentially expressed genes
res <- res[order(res$padj),]
res <- subset(res, res$padj < 0.05)

# Order all differentially expressed genes by effect size (the absolute value of log2FoldChange)
res <- res[order(-abs(res$log2FoldChange)),]
res <- as.data.frame(res)

cols<-!(colnames(res) %in% c("baseMean","lfcSE","stat","pvalue"))
res_subset <-subset(res,,cols)

### Extract genes upregulated
res_up <- res_subset %>% 
  filter(log2FoldChange > 0)
write.table(res_up, file="~/Documents/TFM/GitHub/HB_PublicData/1-Obtaining-DEGs-for-HB/DEGs_HB/Ranks_HB/res_GSE104766_up", sep="\t", quote=F, col.names=TRUE)

### Extract genes downregulated
res_down <- res_subset %>% 
  filter(log2FoldChange < 0)
write.table(res_down, file="~/Documents/TFM/GitHub/HB_PublicData/1-Obtaining-DEGs-for-HB/DEGs_HB/Ranks_HB/res_GSE104766_down", sep="\t", quote=F, col.names=TRUE)
```



# Normalization
```{r}
## Create DSeq2 object
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = ~ tissue)

## Filter out all genes with <10 reads total across all samples
dds <- dds[rowSums(counts(dds)) >= 10,]

## Factor Level
dds$tissue <- relevel(dds$tissue, ref = "Normal")

## Median of ratios Normalization
dds <- estimateSizeFactors(dds)
```


# GSE104766

## Import data
```{r}
# Import Data

# Clean Matrix
cts_GSE104766 <- aggregate(cts_GSE104766[,-1], list(cts_GSE104766$gene_symbol), mean)
rownames(cts_GSE104766) <- cts_GSE104766$Group.1
cts_GSE104766 <- subset( cts_GSE104766, select = -Group.1 )
cts_GSE104766[,]<- lapply(lapply(cts_GSE104766[,-1],round),as.integer)
# Convert data to matrix
cts_GSE104766 <- as.matrix(cts_GSE104766)
```

```{r}
# Import Metadata
Meta_GSE104766_path <- '~/Documents/TFM/GitHub/HB_PublicData/1-Obtaining-DEGs-for-HB/Metadata_HB/GSE104766_filtered_metadata.csv'
coldata_GSE104766 <- read.csv(Meta_GSE104766_path, row.names=1)
coldata_GSE104766$tissue[coldata_GSE104766$tissue == "Normal liver"] <- "Normal"
coldata_GSE104766$tissue <- factor(coldata_GSE104766$tissue)
```

Check that data and metadata are consistent and in the same order
```{r}
all(rownames(coldata_GSE104766) %in% colnames(cts_GSE104766))
all(rownames(coldata_GSE104766) == colnames(cts_GSE104766))
cts_GSE133039 <- cts_GSE133039[, rownames(coldata_GSE133039)]
all(rownames(coldata_GSE133039) == colnames(cts_GSE133039))
```
## Normalization
```{r}
#Create DSeq2 object
dds_GSE104766 <- DESeqDataSetFromMatrix(countData = cts_GSE104766,
                              colData = coldata_GSE104766,
                              design = ~ tissue)

# Filter out all genes with <10 reads total across all samples
dds_GSE104766 <- dds_GSE104766[rowSums(counts(dds_GSE104766)) >= 10,]

# Factor Level
dds_GSE104766$tissue <- relevel(dds_GSE104766$tissue, ref = "Normal")

# Median of ratios Normalization
dds_GSE104766 <- estimateSizeFactors(dds_GSE104766)
#normalized_counts_GSE104766 <- counts(dds_GSE104766, normalized=TRUE)
#write.table(normalized_counts_GSE104766, file="Matrices_HB/GSE104766_normalized_counts.txt", sep="\t", quote=F, col.names=NA)
```

## Find DEGs
```{r}
# Run differential expression analysis
dds_GSE104766 <- DESeq(dds_GSE104766, parallel=TRUE)
res_GSE104766 <- results(dds_GSE104766, lfcThreshold = 1, parallel=TRUE)


# Extract all differentially expressed genes
res_GSE104766 <- res_GSE104766[order(res_GSE104766$padj),]
res_GSE104766 <- subset(res_GSE104766, res_GSE104766$padj < 0.05)

# Order all differentially expressed genes by effect size (the absolute value of log2FoldChange)
res_GSE104766 <- res_GSE104766[order(-abs(res_GSE104766$log2FoldChange)),]
res_GSE104766 <- as.data.frame(res_GSE104766)

cols<-!(colnames(res_GSE104766) %in% c("baseMean","lfcSE","stat","pvalue"))
res_GSE104766_subset <-subset(res_GSE104766,,cols)

### Extract genes upregulated
res_GSE104766_up <- res_GSE104766_subset %>% 
  filter(log2FoldChange > 0)
write.table(res_GSE104766_up, file="~/Documents/TFM/GitHub/HB_PublicData/1-Obtaining-DEGs-for-HB/Matrices_HB/Ranks_HB/res_GSE104766_up", sep="\t", quote=F, col.names=TRUE)

### Extract genes downregulated
res_GSE104766_down <- res_GSE104766_subset %>% 
  filter(log2FoldChange < 0)
write.table(res_GSE104766_down, file="~/Documents/TFM/GitHub/HB_PublicData/1-Obtaining-DEGs-for-HB/Matrices_HB/Ranks_HB/res_GSE104766_down", sep="\t", quote=F, col.names=TRUE)
```



# GSE133039

## Import data
```{r}
# Import Data
cts_GSE133039_path <- "~/Documents/TFM/GitHub/HB_PublicData/1-Obtaining-DEGs-for-HB/Matrices_HB/Originals_HB/GSE133039_GeneLevel_Raw_data.csv"
cts_GSE133039 <- read.csv(cts_GSE133039_path)
# Clean Matrix
cts_GSE133039 <- aggregate(cts_GSE133039[,-1], list(cts_GSE133039$gene_symbol), mean)
rownames(cts_GSE133039) <- cts_GSE133039$Group.1
cts_GSE133039 <- subset( cts_GSE133039, select = -Group.1 )
cts_GSE133039[,]<- lapply(lapply(cts_GSE133039[,-1],round),as.integer)
# Convert data to matrix
cts_GSE133039 <- as.matrix(cts_GSE133039)
```

```{r}
# Import Metadata
Meta_GSE133039_path <- '~/Documents/TFM/GitHub/HB_PublicData/1-Obtaining-DEGs-for-HB/Metadata_HB/GSE133039_filtered_metadata.csv'
coldata_GSE133039 <- read.csv(Meta_GSE133039_path, row.names=1)
coldata_GSE133039$tissue[coldata_GSE133039$tissue == "Normal liver tissue"] <- "Normal"
coldata_GSE133039$tissue[coldata_GSE133039$tissue == "Tumor liver tissue"] <- "Hepatoblastoma"
coldata_GSE133039$tissue <- factor(coldata_GSE133039$tissue)
```

Check that data and metadata are consistent and in the same order
```{r}
all(rownames(coldata_GSE133039) %in% colnames(cts_GSE133039))
all(rownames(coldata_GSE133039) == colnames(cts_GSE133039))
#cts_GSE133039 <- cts_GSE133039[, rownames(coldata_GSE133039)]
#all(rownames(coldata_GSE133039) == colnames(cts_GSE133039))
```
## Normalization
```{r}
#Create DSeq2 object
dds_GSE133039 <- DESeqDataSetFromMatrix(countData = cts_GSE133039,
                              colData = coldata_GSE133039,
                              design = ~ tissue)

# Filter out all genes with <10 reads total across all samples
dds_GSE133039 <- dds_GSE133039[rowSums(counts(dds_GSE133039)) >= 10,]

# Factor Level
dds_GSE133039$tissue <- relevel(dds_GSE133039$tissue, ref = "Normal")

# Median of ratios Normalization
dds_GSE133039 <- estimateSizeFactors(dds_GSE133039)
#normalized_counts_GSE104766 <- counts(dds_GSE104766, normalized=TRUE)
#write.table(normalized_counts_GSE104766, file="Matrices_HB/GSE104766_normalized_counts.txt", sep="\t", quote=F, col.names=NA)
```

## Find DEGs
```{r}
# Run differential expression analysis
dds_GSE133039 <- DESeq(dds_GSE133039, parallel=TRUE)
res_GSE133039 <- results(dds_GSE133039, lfcThreshold = 1, parallel=TRUE)


# Extract all differentially expressed genes
res_GSE133039 <- res_GSE133039[order(res_GSE133039$padj),]
res_GSE133039 <- subset(res_GSE133039, res_GSE133039$padj < 0.05)

# Order all differentially expressed genes by effect size (the absolute value of log2FoldChange)
res_GSE133039 <- res_GSE133039[order(-abs(res_GSE133039$log2FoldChange)),]
res_GSE133039 <- as.data.frame(res_GSE133039)

cols<-!(colnames(res_GSE133039) %in% c("baseMean","lfcSE","stat","pvalue"))
res_GSE133039_subset <-subset(res_GSE133039,,cols)

### Extract genes upregulated 
res_GSE133039_up <- res_GSE133039_subset %>% 
  filter(log2FoldChange > 0)
write.table(res_GSE133039_up, file="~/Documents/TFM/GitHub/HB_PublicData/1-Obtaining-DEGs-for-HB/Matrices_HB/Ranks_HB/res_GSE133039_up", sep="\t", quote=F, col.names=TRUE)

### Extract genes downregulated 
res_GSE133039_down <- res_GSE133039_subset %>% 
  filter(log2FoldChange < 0)
write.table(res_GSE133039_down, file="~/Documents/TFM/GitHub/HB_PublicData/1-Obtaining-DEGs-for-HB/Matrices_HB/Ranks_HB/res_GSE133039_down", sep="\t", quote=F, col.names=TRUE)
```


# GSE89775
https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE133039

## Count matrix input
```{r}
# Import Data
cts_GSE89775_path <- "Matrices_HB/Originals_HB/GSE89775_GeneLevel_Raw_data.csv"
cts_GSE89775 <- read.csv(cts_GSE89775_path,sep=",", quote="", row.names=1)
cts_GSE89775[,]<- lapply(lapply(cts_GSE89775[,-1],round),as.integer)
```

```{r}
# Import Metadata
Meta_GSE89775_path <- 'Metadata_HB/GSE89775_filtered_metadata.csv'
coldata_GSE89775 <- read.csv(Meta_GSE89775_path, row.names=1)

coldata_GSE89775$tissue[coldata_GSE89775$tissue == "Hepatoblastoma liver"] <- "Hepatoblastoma"
coldata_GSE89775$tissue[coldata_GSE89775$tissue == "Normal liver"] <- "Normal"
coldata_GSE89775$tissue <- factor(coldata_GSE89775$tissue)
```


```{r}
# Convert data to matrix
cts_GSE89775 <- as.matrix(cts_GSE89775)
```

Check that data and metadata are consistent and in the same order
```{r}
all(rownames(coldata_GSE89775) %in% colnames(cts_GSE89775))
all(rownames(coldata_GSE89775) == colnames(cts_GSE89775))
cts_GSE89775<- cts_GSE89775[, rownames(coldata_GSE89775)]
all(rownames(coldata_GSE89775) == colnames(cts_GSE89775))
```

## Normalization
Create DSeq2 object
```{r}
dds_GSE89775 <- DESeqDataSetFromMatrix(countData = cts_GSE89775,
                              colData = coldata_GSE89775,
                              design = ~ tissue)
```

### Pre-filtering
While it is not necessary to pre-filter low count genes before running the DESeq2 functions, there are two reasons which make pre-filtering useful: by removing rows in which there are very few reads, we reduce the memory size of the dds data object, and we increase the speed of the transformation and testing functions within DESeq2. Here we perform a minimal pre-filtering to keep only rows that have at least 10 reads total.
```{r}
keep <- rowSums(counts(dds_GSE89775)) >= 10
dds_GSE89775 <- dds_GSE89775[keep,]
```

### Factor Level
```{r}
dds_GSE89775$tissue <- relevel(dds_GSE89775$tissue, ref = "Normal")
```

### Median of ratios Normalization
https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html
```{r}
dds_GSE89775 <- estimateSizeFactors(dds_GSE89775)
#normalized_counts_GSE89775 <- counts(dds_GSE89775, normalized=TRUE)
#write.table(normalized_counts_GSE89775, file="Matrices_HB/GSE89775_normalized_counts.txt", sep="\t", quote=F, col.names=NA)
```

## Find DEGs
```{r}
dds_GSE89775 <- DESeq(dds_GSE89775, parallel=TRUE)
res_GSE89775 <- results(dds_GSE89775, parallel=TRUE)
```
```{r}
resSig_GSE89775 <- subset(res_GSE89775, padj < 0.05)
resSig_GSE89775 <- subset(resSig_GSE89775, abs(log2FoldChange) > 1)
nrow(resSig_GSE89775)
resSig_GSE89775 <- as.data.frame(resSig_GSE89775)
```
```{r}
resSig_GSE89775 <- resSig_GSE89775[ order( -resSig_GSE89775$log2FoldChange ), ]
cols<-!(colnames(resSig_GSE89775) %in% c("baseMean","lfcSE","stat","pvalue","padj"))
resSig_GSE89775_subset <-subset(resSig_GSE89775,,cols)
 
write.table(resSig_GSE89775_subset, file="Matrices_HB/Ranks_HB/GSE89775_ranks.txt", sep="\t", quote=F, col.names=TRUE)
```

# GSE151347
https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE133039

## Count matrix input
```{r}
# Import Data
cts_GSE151347_path <- "Matrices_HB/Originals_HB/GSE151347_GeneLevel_Raw_data.csv"
cts_GSE151347 <- read.csv(cts_GSE151347_path,sep=",", quote="", row.names=1)
cts_GSE151347[,]<- lapply(lapply(cts_GSE151347[,-1],round),as.integer)
```

```{r}
# Import Metadata
Meta_GSE151347_path <- 'Metadata_HB/GSE151347_filtered_metadata.csv'
coldata_GSE151347 <- read.csv(Meta_GSE151347_path, row.names=1)
names(coldata_GSE151347)[names(coldata_GSE151347) == "tissue.type"] <- "tissue"
coldata_GSE151347$tissue[coldata_GSE151347$tissue == "Normal liver"] <- "Normal"
coldata_GSE151347$tissue <- factor(coldata_GSE151347$tissue)
```

```{r}
# Convert data to matrix
cts_GSE89775 <- as.matrix(cts_GSE151347)
```

Check that data and metadata are consistent and in the same order
```{r}
all(rownames(coldata_GSE151347) %in% colnames(cts_GSE151347))
all(rownames(coldata_GSE151347) == colnames(cts_GSE151347))
cts_GSE151347 <- cts_GSE151347[, rownames(coldata_GSE151347)]
all(rownames(coldata_GSE151347) == colnames(cts_GSE151347))
```
## Normalization
Create DSeq2 object
```{r}
dds_GSE151347 <- DESeqDataSetFromMatrix(countData = cts_GSE151347,
                              colData = coldata_GSE151347,
                              design = ~ tissue)
```

### Pre-filtering
While it is not necessary to pre-filter low count genes before running the DESeq2 functions, there are two reasons which make pre-filtering useful: by removing rows in which there are very few reads, we reduce the memory size of the dds data object, and we increase the speed of the transformation and testing functions within DESeq2. Here we perform a minimal pre-filtering to keep only rows that have at least 10 reads total.
```{r}
keep <- rowSums(counts(dds_GSE151347)) >= 10
dds_GSE151347 <- dds_GSE151347[keep,]
```

### Factor Level
```{r}
dds_GSE151347$tissue <- relevel(dds_GSE151347$tissue, ref = "Normal")
```

### Median of ratios Normalization
https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html
```{r}
dds_GSE151347 <- estimateSizeFactors(dds_GSE151347)
normalized_counts_GSE151347 <- counts(dds_GSE151347, normalized=TRUE)
write.table(normalized_counts_GSE151347, file="Matrices_HB/Ranks_HB/GSE151347_normalized_counts.txt", sep="\t", quote=F, col.names=NA)
```

## Find DEGs
```{r}
dds_GSE151347 <- DESeq(dds_GSE151347, parallel=TRUE)
res_GSE151347 <- results(dds_GSE151347, parallel=TRUE)
```
```{r}
resSig_GSE151347 <- subset(res_GSE151347, padj < 0.05)
resSig_GSE151347 <- subset(resSig_GSE151347, abs(log2FoldChange) > 1)
nrow(resSig_GSE151347)
resSig_GSE151347 <- as.data.frame(resSig_GSE151347)
```
```{r}
resSig_GSE151347 <- resSig_GSE151347[ order( -resSig_GSE151347$log2FoldChange ), ]
cols<-!(colnames(resSig_GSE151347) %in% c("baseMean","lfcSE","stat","pvalue","padj"))
resSig_GSE151347_subset <-subset(resSig_GSE151347,,cols)
 
write.table(resSig_GSE151347_subset, file="Matrices_HB/GSE151347_ranks.txt", sep="\t", quote=F, col.names=TRUE)
```



